auction-app-frontend/src/components/Header/Header.jsx

```diff
@@ -1,23 +1,43 @@
+import { Link, useNavigate } from "react-router-dom";
+
 import { SocialMediaIcons } from "src/components";
 
+import { ROUTE_PATHS } from "src/constants";
+import { logoutUser } from "src/services/userService";
+import { useUserName } from "src/store/UserNameContext";
+
 import "./style.scss";
 
 const Header = () => {
-  // hardcoded for now
-  const isLoggedIn = false;
-  const userName = "John Doe";
+  const { userName, setUserName } = useUserName();
+  const navigate = useNavigate();
 
+  const onLogout = () => {
+    setUserName(null);
+    localStorage.removeItem('userName');
```

Comment:
just add some space between different logical units, state and local storage should be separated